esp/esp/program/modules/handlers/teacherclassregmodule.py

```diff
@@ -317,6 +304,79 @@ def process_attendance(section, request, prog):
+                sections = ClassSection.objects.filter(id=request.POST['secid'])
+                if not sections.exists():
+                    json_data['message'] = 'Section not found!'
+                else:
+                    section = sections[0]
+                    json_data['secid'] = section.id
+                    if request.POST.get('undo', 'false').lower() == 'true':
+                        #should we also delete the program attendance record?
+                        srs = StudentRegistration.valid_objects().filter(user = student, section = section, relationship = attended)
+                        if srs.exists():
+                            for sr in srs:
+                                sr.expire() #or delete??
+                            json_data['message'] = '%s is no longer marked as attending.' % student.name()
+                        else:
+                            json_data['message'] = '%s was not marked as attending.' % student.name()
+                    else:
+                        if not prog.isCheckedIn(student):
+                            rec = Record(user=student, program=prog, event='attended')
+                            rec.save()
+                            json_data['checkedin'] = True
+                        sr = StudentRegistration.objects.get_or_create(user = student, section = section, relationship = attended, start_date__range=(today_min, today_max))[0]
+                        sr.end_date = today_max
+                        sr.save()
+                        if student not in section.students():
+                            if request.POST.get('unenroll', 'true').lower() == 'true':
+                                sm = ScheduleMap(student, prog)
+                                for ts in [ts.id for ts in section.get_meeting_times()]:
+                                    if ts in sm.map and len(sm.map[ts]) > 0:
+                                        for sm_sec in sm.map[ts]:
+                                            sm_sec.unpreregister_student(student)
+                            if request.POST.get('enroll', 'true').lower() == 'true':
+                                for rt in [enrolled, onsite]:
+                                    srs = StudentRegistration.objects.filter(user = student, section = section, relationship = rt)
+                                    if srs.count() > 0:
+                                        sr = srs[0]
+                                        sr.unexpire()
+                                    else:
+                                        sr = StudentRegistration.objects.create(user = student, section = section, relationship = rt)
+                                    if rt.name=='OnSite/AttendedClass':
+                                        sr.end_date = today_max
+                                        sr.save()
+                        json_data['message'] = '%s is marked as attending.' % student.name()
```

Comment:
Honestly, I don't think these status messages are necessary. I think the red disable hover and the "updating attendance" are useful to make the user understand that something is happening, but I feel like unless something went wrong, the check/uncheck symbolizes everything and the status is redundant. With a long list of students, the status messages appearing for every attending student will get super annoying

Reply:
My other thought is to have the status messages appear for 10 seconds and then disappear but I think it's too much work and they aren't necessary anyways